isort/deprecated/finders.py

```diff
@@ -29,7 +29,7 @@
     from pip_api import parse_requirements  # type: ignore
 
 except ImportError:
-    parse_requirements = None
+    parse_requirements = None  # type: ignore[assignment]
```

Comment:
This can be fixed by initialising `parse_requirements` on line 27 as `parse_requirements: types.ModelType | None` right? Or does `mypy` no longer support that pattern?

Reply:
> This can be fixed by initialising `parse_requirements` on line 27 as `parse_requirements: types.ModelType | None` right? Or does `mypy` no longer support that pattern?

Sorry, what is `types.ModelType`?

I think the typing would have to be something like (from a glance at https://github.com/di/pip-api/blob/21eea42e37b29e18c004a922f8989108a006a203/pip_api/_parse_requirements.py#L484):

```python
parse_requirements: (
    Callable[
        [os.PathLike[str], Optional[Any], bool, bool],
        Dict[str, Union[pip_api.Requirement, pip_api.UnparsedRequirement]],
    ]
    | None
)
```

but that's a bit tricky, because at this point `pip_api` isn't imported, so `pip_api.UnparsedRequirement` is unknown